code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+//I JUST WANNA GRILL FOR GOD'S SAKE
+
+/obj/machinery/grill
+	name = "grill"
+	desc = "Just like the old days."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "grill_open"
+	density = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilled_item
```

Comment:
Think about what this should do on Destroy(). Also clear the reference on Exited().

Reply:
does destroy not empty the contents

Reply:
Yes but it doesn't null this variable.

Reply:
i mean its destroyed so not like it will be used